### ISSUE TITLE
Pipeline crash if timeout_timestamp_field is missing from event

### ISSUE THREAD

**Issue by TheVastyDeep** [logstash-plugins/logstash-filter-aggregate#121]
logstash 8.8.1 installed on Linux via RPM and run on the command line. aggregate is bundled with the RPM. Bundled JVM used.

If the timeout_timestamp_field is missing from an event then the pipeline crashes [here](https://github.com/logstash-plugins/logstash-filter-aggregate/blob/e216a60fb96cfe679c28210747838ea36ef067d8/lib/logstash/filters/aggregate.rb#L477) with the error.

> [ERROR][logstash.javapipeline    ][main] Pipeline worker error, the pipeline will be stopped {:pipeline_id=>"main", :error=>"(NoMethodError) undefined method `time' for nil:NilClass",

Reproducible with

    input { generator { count => 1 lines => [ '1' ] } }
    output { stdout { codec => rubydebug { metadata => false } } }
    filter {
            aggregate {
                task_id => "%{message}"
                timeout_timestamp_field => "foo"
                timeout => 5
                push_map_as_event_on_timeout => true
            }
        }
    }

**Comment by fbaligand** [comment]
Well, this option requires that timestamp field exists on each event. Or else this option can’t do its job.

**Comment by TheVastyDeep** [comment]
Yes, that is obviously true, but it would be trivial to produce a clearer error message. and arguably a pipeline crash is not the optimal outcome.

**Comment by fbaligand** [comment]
Yes, I agree with you.
